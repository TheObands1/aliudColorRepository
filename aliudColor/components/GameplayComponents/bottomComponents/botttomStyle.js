import { StyleSheet } from "react-native";

export default StyleSheet.create({
    counterCount: {
        fontFamily: 'dogbyte',
        textAlign: 'center',
        color: "#ecf0f1",
        fontSize: 50
      },
      counterLabel: {
        fontFamily: 'dogbyte',
        textAlign: 'center',
        color: '#bbb',
        fontSize: 20,
        textTransform: "uppercase"
      },
      bestScoreContainer: {
        marginTop: 10,
        flexDirection: 'row',
        justifyContent: 'center'
      },
      bestScoreIcon: {
        width: 25,
        height: 25,
        marginRight: 5
      },
      bestScoreLabel: {
        fontFamily: 'dogbyte',
        color: '#bbb',
        fontSize: 25,
        marginTop: 2.5,
      },
      gameStateIcon: {
        width: 50, 
        height: 50
      }
});
