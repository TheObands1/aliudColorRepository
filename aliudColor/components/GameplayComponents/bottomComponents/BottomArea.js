import React from "react";
import { Text, View, StyleSheet, TouchableOpacity,  Dimensions, Image} from "react-native";
import botttomStyle from "./botttomStyle";
import PropTypes from "prop-types";
import * as SQLite from 'expo-sqlite';

const BottomArea = ({gameState, changeGameState, points, timeLeft, windowWidth, goBackToMainMenu, currentMaxPoints}) => {
    
    const gameDatabase = SQLite.openDatabase("gameDatabase");
    const gameStateButtonIcon = gameState === "Playing" ? require("../../../assets/icons/pause.png")
                                                : gameState === "Paused" ? require("../../../assets/icons/play.png")
                                                : require("../../../assets/icons/replay.png");
                
    function showHomeButton(){
        if(gameState === "Lost" || gameState === "Paused")
        {
            return (<TouchableOpacity style={{ alignItems: 'center', marginTop: 10 }} onPress={goBackToMainMenu} 
                        accessibilityLabel="Go Home Button"
                        accessibilityHint="Takes you back to the main menu">
                        <Image source={require("../../../assets/icons/home.png")} style={botttomStyle.gameStateIcon} />
                    </TouchableOpacity>);
        }
        else
        {
            return;
        }
    }

    return (
        <View style={{flex: 1, width: windowWidth * 0.875, flexDirection: 'row',}}>
            <View style={{ flex: 1, marginTop: 'auto', marginBottom: 'auto'}}> 
                <View accessible={true}>
                    <Text style={botttomStyle.counterCount}>{points}</Text>
                    <Text style={botttomStyle.counterLabel}>points</Text>
                </View>
                <View style={botttomStyle.bestScoreContainer} accessible={true} accessibilityLabel= {"Current High Score"+currentMaxPoints}>
                    <Image source={require("../../../assets/icons/trophy.png")} style={botttomStyle.bestScoreIcon}/>
                    <Text style={botttomStyle.bestScoreLabel}>{currentMaxPoints}</Text>
                </View>
            </View>

            <View style={{ flex: 1, marginTop: 'auto', marginBottom: 'auto'}}>
                <TouchableOpacity style={{ alignItems: 'center' }} 
                    onPress={changeGameState}
                    accessibilityLabel= { gameState === "Playing" ? "Pause Game Button"
                                                : gameState === "Paused" ? "Resume Game Button"
                                                : "Replay Button"}>
                    <Image source={gameStateButtonIcon} style={botttomStyle.gameStateIcon} />
                </TouchableOpacity>
                {
                    showHomeButton()
                }
            </View>

            <View style={{ flex: 1, marginTop: 'auto', marginBottom: 'auto'}}>
                <View accessible={true}>
                    <Text style={botttomStyle.counterCount}>{timeLeft}</Text>
                    <Text style={botttomStyle.counterLabel}>time left</Text>
                </View>
                <View style={botttomStyle.bestScoreContainer} accessibilityLabel="Time Icon">
                    <Image source={require("../../../assets/icons/clock.png")} style={botttomStyle.bestScoreIcon}/>
                </View>
            </View>
        </View>
    );
}

export { BottomArea };