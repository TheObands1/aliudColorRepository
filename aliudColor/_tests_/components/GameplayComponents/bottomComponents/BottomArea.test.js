import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

import {BottomArea} from "../../../../components/GameplayComponents/bottomComponents/BottomArea"

describe('tests for the <BottomArea/>', () => {
    
it('Render the component correctly ', () => {
    const tree = renderer.create(<BottomArea gameState = {"Lost"}
                                            changeGameState = {()=>console.log("Test")}
                                            points = {0}
                                            timeLeft = {0}
                                            windowWidth = {1}
                                            goBackToMainMenu = {()=>console.log("Test")}
                                            currentMaxPoints = {0}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  });