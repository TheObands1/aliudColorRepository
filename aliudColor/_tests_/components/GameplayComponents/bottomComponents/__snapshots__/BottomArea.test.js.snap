// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the <BottomArea/> Render the component correctly  1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "width": 0.875,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": "auto",
        "marginTop": "auto",
      }
    }
  >
    <View
      accessible={true}
    >
      <Text
        style={
          Object {
            "color": "#ecf0f1",
            "fontFamily": "dogbyte",
            "fontSize": 50,
            "textAlign": "center",
          }
        }
      >
        0
      </Text>
      <Text
        style={
          Object {
            "color": "#bbb",
            "fontFamily": "dogbyte",
            "fontSize": 20,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        points
      </Text>
    </View>
    <View
      accessibilityLabel="Current High Score0"
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginRight": 5,
            "width": 25,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#bbb",
            "fontFamily": "dogbyte",
            "fontSize": 25,
            "marginTop": 2.5,
          }
        }
      >
        0
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": "auto",
        "marginTop": "auto",
      }
    }
  >
    <View
      accessibilityLabel="Replay Button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessibilityHint="Takes you back to the main menu"
      accessibilityLabel="Go Home Button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "marginTop": 10,
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": "auto",
        "marginTop": "auto",
      }
    }
  >
    <View
      accessible={true}
    >
      <Text
        style={
          Object {
            "color": "#ecf0f1",
            "fontFamily": "dogbyte",
            "fontSize": 50,
            "textAlign": "center",
          }
        }
      >
        0
      </Text>
      <Text
        style={
          Object {
            "color": "#bbb",
            "fontFamily": "dogbyte",
            "fontSize": 20,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        time left
      </Text>
    </View>
    <View
      accessibilityLabel="Time Icon"
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginRight": 5,
            "width": 25,
          }
        }
      />
    </View>
  </View>
</View>
`;
